picoquic/util.c

```diff
@@ -458,19 +463,32 @@ int picoquic_get_input_path(char * target_file_path, size_t file_path_max, const
 }
 
 /* Safely open files in a portable way */
-FILE * picoquic_file_open(char const * file_name, char const * flags)
+FILE * picoquic_file_open_ex(char const * file_name, char const * flags, int * last_err)
 {
     FILE * F;
 
 #ifdef _WINDOWS
     errno_t err = fopen_s(&F, file_name, flags);
-    if (err != 0 && F != NULL) {
+    if (err != 0){
+        if (last_err != NULL) {
+            *last_err = err;
+        }
+        if (F != NULL) {
             fclose(F);
             F = NULL;
         }
+    }
 #else
     F = fopen(file_name, flags);
+    if (F == NULL && last_err != NULL) {
+        *last_err = errno;
+    }
 #endif
 
     return F;
+}
+
+FILE* picoquic_file_open(char const* file_name, char const* flags)
```

Comment:
Does C support default initialized parameters? If so, we could do
```
FILE * picoquic_file_open(char const * file_name, char const * flags, int * last_err = NULL)
```

Reply:
I don't think that would fly through all C compilers...

Reply:
Yep. Tried it, Visual Studio already complains :)